benches/transform/route.rs

```diff
@@ -0,0 +1,185 @@
+                r#"
+            route.a.type = "vrl"
+            route.a.source = '.a == "aaa"'
+        "#,
+            )
+            .unwrap(),
+            output_buffer: output_buffer.clone(),
+        },
+        // A small filter where a sole field is mapped into a named route, does
+        // not matche.
+        Param {
+            slug: "vrl_field_not_match",
+            input: event.clone(),
+            route_config: toml::from_str::<RouteConfig>(
+                r#"
+            route.a.type = "vrl"
+            route.a.source = '.a == "aaaaaaa"'
+        "#,
+            )
+            .unwrap(),
+            output_buffer: output_buffer.clone(),
+        },
+        // A larger filter where many fields are mapped, some multiple times,
+        // into named filters. A mixture of match and not match happens.
+        Param {
+            slug: "vrl_field_match_many",
+            input: event.clone(),
+            route_config: toml::from_str::<RouteConfig>(
+                r#"
+            route.a.type = "vrl"
+            route.a.source = '.a == "aaaaaaa"'
+
+            route.b.type = "vrl"
+            route.b.source = '.b == "b"'
+
+            route.c.type = "vrl"
+            route.c.source = '.c == "cccc"'
+
+            route.d.type = "vrl"
+            route.d.source = '.d == "d"'
+
+            route.e.type = "vrl"
+            route.e.source = '.d == "d"'
+
+            route.f.type = "vrl"
+            route.f.source = '.e == "eeeeeeeeee"'
+
+            route.g.type = "vrl"
+            route.g.source = '.f == "f"'
+
+            route.h.type = "vrl"
+            route.h.source = '.f == "qq"'
+
+            route.i.type = "vrl"
+            route.i.source = '.f == "fx"'
+
+            route.j.type = "vrl"
+            route.j.source = '.f == "lf"'
+
+            route.k.type = "vrl"
+            route.k.source = '.f == "fpioalkjasdf"'
+
+            route.l.type = "vrl"
+            route.l.source = '.f == "lkjiouasodifjlkjasdfoiuf"'
+
+            route.m.type = "vrl"
+            route.m.source = '.f == "aaaaf"'
+
+            route.n.type = "vrl"
+            route.n.source = '.f == "0124"'
+
+            route.ay.type = "vrl"
+            route.ay.source = '.a == "0_0"'
+        "#,
+            )
+            .unwrap(),
+            output_buffer: output_buffer.clone(),
+        },
+    ] {
+        group.throughput(Throughput::Elements(param.input.len() as u64));
+        group.bench_with_input(BenchmarkId::new("transform", param), &param, |b, param| {
+            b.iter_batched(
+                || {
+                    let route =
+                        Route::new(&param.route_config.clone(), &TransformContext::default())
+                            .unwrap();
+                    (route, param.input.clone(), param.output_buffer.clone())
+                },
+                |(mut route, input, mut output_buffer)| {
+                    route.transform(input, &mut output_buffer);
```

Comment:
It seems very unlikely that this will be optimized away, but I might wrap `output_buffer` in `criterion::black_box` just-in-case (or return it to the closure).

Reply:
Addressed in 89d4ecc79